colcon_core/command.py

```diff
@@ -177,6 +177,11 @@ def _main(*, command_name, argv):
             None, None)
         handler.handle(log_record)
 
+    # set an environment variable named after the command (if not already set)
+    # which allows subprocesses to identify they are invoked by this command
+    if command_name.upper() not in os.environ:
+        os.environ[command_name.upper()] = '1'
```

Comment:
What does this do if colcon is invoked using `python3 -m colcon`?

Reply:
The `command_name` should still be `colcon`: https://github.com/colcon/colcon-core/blob/aae749ce6c42199632631e37ca57f1653c2f5fd7/colcon_core/command.py#L78